judge/judgedaemon.main.php

```diff
@@ -70,39 +92,38 @@ function request(string $url, string $verb = 'GET', string $data = '', bool $fai
         $lastrequest = $url;
     }
 
+    setup_curl_handle();
+
     $url = $resturl . "/" . $url;
     if ($verb == 'GET') {
         $url .= '?' . $data;
     }
 
-    $ch = curl_init($url);
-    curl_setopt($ch, CURLOPT_USERAGENT, "DOMjudge/" . DOMJUDGE_VERSION);
-    curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
-    curl_setopt($ch, CURLOPT_USERPWD, $restuser . ":" . $restpass);
-    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($curl_handle, CURLOPT_URL, $url);
+
     if ($verb == 'POST') {
-        curl_setopt($ch, CURLOPT_POST, true);
+        curl_setopt($curl_handle, CURLOPT_POST, true);
```

Comment:
Adding 
`curl_setopt($curl_handle, CURLOPT_POST, false);`
two lines above gets past the second request. It still ends up with 404 later, but I'll keep digging what else we have to reset.